app/gist/route.js

```diff
@@ -26,6 +26,25 @@ export default Ember.Route.extend({
       this.get('controller').set('unsaved', false);
     },
 
+    fork (gist) {
+      gist.fork().then((response) => {
+        let id = response.id;
+        let newGist = this.duplicate(gist);
+        return newGist.save().then(() => {
+          this.transitionTo('gist.edit', id);
+        });
+      }).catch((error) => {
+        if (error && error.errors) {
+          let firstError = error.errors[0];
```

Comment:
extracting the error callback into its own function, so it can be united tested will help cleanup this code some more. The  extra verbosity distracts from the overall flow of data.